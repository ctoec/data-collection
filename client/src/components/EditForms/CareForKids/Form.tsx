import React, { useState, useContext } from 'react';
import {
  Form,
  FormSubmitButton,
<<<<<<< HEAD
  FormField,
  RadioButtonGroupProps,
  RadioButtonGroup,
  RadioButton,
} from '@ctoec/component-library';
import { Child } from '../../../shared/models';
import { apiPut } from '../../../utils/api';
import AuthenticationContext from '../../../contexts/AuthenticationContext/AuthenticationContext';
import { EditFormProps } from '../types';
=======
} from '@ctoec/component-library';
import { CareForKidsField } from './CareForKidsField';
import { EditFormProps } from '../types';
import AuthenticationContext from '../../../../contexts/AuthenticationContext/AuthenticationContext';
import { Child } from '../../../../shared/models';
import { apiPut } from '../../../../utils/api';
>>>>>>> 262909d... Align headers

/*
 * Basic functional component designed to allow user to edit
 * the Care For Kids field of a Child object.
 */
export const CareForKidsForm: React.FC<EditFormProps> = ({
  child,
  onSuccess,
}) => {
  const { accessToken } = useContext(AuthenticationContext);
  const [isSaving, setIsSaving] = useState(false);

  if (!child) return <></>;

  const onSubmit = (updatedChild: Child) => {
    setIsSaving(true);
    apiPut(`children/${child.id}`, updatedChild, { accessToken })
      .then(() => onSuccess())
      .catch((err) => {
        console.log(err);
      })
      .finally(() => setIsSaving(false));
  };

  return (
    <div className="grid-container margin-top-2">
<<<<<<< HEAD
      <h2 className="grid-row">Receiving Care For Kids?</h2>
=======
      <h2 className="grid-row">Care 4 Kids</h2>
>>>>>>> 262909d... Align headers
      <div>
        <Form<Child>
          className="CareForKidsForm"
          data={child}
          onSubmit={onSubmit}
          noValidate
          autoComplete="off"
        >
<<<<<<< HEAD
          <FormField<Child, RadioButtonGroupProps, boolean>
            getValue={(data) => data.at('recievesC4K')}
            preprocessForDisplay={(data) => (data === true ? 'Yes' : 'No')}
            parseOnChangeEvent={(e) => {
              return e.target.value === 'Yes';
            }}
            inputComponent={RadioButtonGroup}
            id="c4k-radio-group"
            name="careforkids"
            legend="receives care for kids"
            options={[
              {
                render: (props) => (
                  <div>
                    <RadioButton text="Yes" {...props} />
                  </div>
                ),
                value: 'Yes',
              },
              {
                render: (props) => (
                  <div>
                    <RadioButton text="No" {...props} />
                  </div>
                ),
                value: 'No',
              },
            ]}
          />
=======
          <CareForKidsField />
>>>>>>> 262909d... Align headers
          <div className="grid-row margin-top-2">
            <FormSubmitButton text={isSaving ? 'Saving...' : 'Save'} />
          </div>
        </Form>
      </div>
    </div>
  );
};
