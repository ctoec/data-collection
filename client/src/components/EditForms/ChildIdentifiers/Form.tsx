import React, { useContext, useState } from 'react';
import {
  SasidField,
  FirstNameField,
  MiddleNameField,
  LastNameField,
  SuffixField,
  DateOfBirthField,
  BirthCertificateFormFieldSet,
} from './Fields';
import { Form, FormSubmitButton } from '@ctoec/component-library';
import { EditFormProps } from '../types';
import AuthenticationContext from '../../../contexts/AuthenticationContext/AuthenticationContext';
import { Child } from '../../../shared/models';
import { apiPut } from '../../../utils/api';
import useIsMounted from '../../../hooks/useIsMounted';

// import { useFocusFirstError } from '../../../hooks/useFocusFirstError';

export const ChildIdentifiersForm = ({
  child,
  onSuccess,
  hideHeader = false,
}: EditFormProps) => {
  const { accessToken } = useContext(AuthenticationContext);
  const isMounted = useIsMounted();
  const [saving, setSaving] = useState(false);
  // TODO: HOW ARE WE HANDLING ERRORS?
  // const [error, setError] = useState(null);

  // Focus should automatically be on first error on page
  // useFocusFirstError([error]);

  if (!child) {
    throw new Error('Child info rendered without child');
  }

  const onFormSubmit = (_child: Child) => {
    setSaving(true);
    apiPut(`children/${child.id}`, _child, { accessToken })
      .then(() => onSuccess())
      .catch((err) => {
        console.log(err);
      })
      .finally(() => (isMounted() ? setSaving(false) : null));
  };

  return (
    <Form<Child>
      className="ChildIdentifiersForm usa-form"
      data={child}
      onSubmit={onFormSubmit}
      noValidate
      autoComplete="off"
    >
      {!hideHeader && (
          <h2>Child's identifiers</h2>
      )}
      <div className="grid-row grid-gap">
        <div className="mobile-lg:grid-col-12">
          <SasidField />
        </div>
        <div className="mobile-lg:grid-col-9">
          <FirstNameField />
        </div>
        <div className="mobile-lg:grid-col-9">
          <MiddleNameField />
        </div>
        <div className="display-flex flex-row flex-align-end grid-row grid-gap">
          <div className="mobile-lg:grid-col-9">
            <LastNameField />
          </div>
          <div className="mobile-lg:grid-col-3">
            <SuffixField />
          </div>
        </div>
      </div>
      <DateOfBirthField />
      <BirthCertificateFormFieldSet />
      <FormSubmitButton
        text={saving ? 'Saving...' : 'Save'}
        disabled={saving}
      />
    </Form>
  );
};
