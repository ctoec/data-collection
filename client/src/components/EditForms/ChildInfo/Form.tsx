import React, { useContext, useState, useEffect } from 'react';
import {
  RaceField,
  EthnicityField,
  GenderField,
  FosterCheckbox,
} from './Fields';
import { Form, FormSubmitButton } from '@ctoec/component-library';
import { EditFormProps } from '../types';
import AuthenticationContext from '../../../contexts/AuthenticationContext/AuthenticationContext';
import { Child } from '../../../shared/models';
import { apiPut } from '../../../utils/api';
import useIsMounted from '../../../hooks/useIsMounted';

// import { useFocusFirstError } from '../../../hooks/useFocusFirstError';

export const ChildInfoForm = ({
  child,
  onSuccess,
  setAlerts,
  hideHeader = false,
}: EditFormProps) => {
  const { accessToken } = useContext(AuthenticationContext);
  const isMounted = useIsMounted();
  const [saving, setSaving] = useState(false);
  // TODO: HOW ARE WE HANDLING ERRORS?
  // const [error, setError] = useState(null);

  // Focus should automatically be on first error on page
  // useFocusFirstError([error]);

  // Clear any previously displayed alerts from other tabs
  useEffect(() => {
    setAlerts([]);
  }, []);

  if (!child) {
    throw new Error('Child info rendered without child');
  }

  const onFormSubmit = (_child: Child) => {
    setSaving(true);
    apiPut(`children/${child.id}`, _child, { accessToken })
      .then(() => onSuccess())
      .catch((err) => {
        console.log(err);
      })
      .finally(() => (isMounted() ? setSaving(false) : null));
  };

  return (
    <Form<Child>
      className="ChildInfoForm usa-form"
      data={child}
      onSubmit={onFormSubmit}
      noValidate
      autoComplete="off"
    >
      {!hideHeader && (
        <>
          <h2>Child info</h2>
          <p className="usa-hint">
            Information is required unless otherwise specified.
          </p>
        </>
      )}
      <RaceField />
      <EthnicityField />
      <GenderField />
      <FosterCheckbox />
      <FormSubmitButton
        text={saving ? 'Saving...' : 'Save'}
        disabled={saving}
      />
    </Form>
  );
};
