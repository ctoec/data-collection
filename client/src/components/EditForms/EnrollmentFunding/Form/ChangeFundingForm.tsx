import React, { useContext, useState } from 'react';
import {
  Enrollment,
  FundingSpace,
  ReportingPeriod,
} from '../../../../shared/models';
import AuthenticationContext from '../../../../contexts/AuthenticationContext/AuthenticationContext';
import { ChangeFunding } from '../../../../shared/payloads/ChangeFunding';
import { apiPost } from '../../../../utils/api';
import {
  Card,
  Button,
  Form,
  FormSubmitButton,
  Alert,
} from '@ctoec/component-library';
import { ReportingPeriodField } from '../Fields/Funding/ReportingPeriod';
import { FundingField } from '../Fields/Funding/NewFunding';

type ChangeFundingFormProps = {
  fundingSpaces: FundingSpace[];
  reportingPeriods: ReportingPeriod[];
  enrollment: Enrollment;
  onSuccess: () => void;
};

/**
 * Component for gathering user input to change enrollment's funding.
 * Uses a ChangeFunding data object to enable the user to provide
 * new funding data and/or last reporting period for previously current
 * funding, based on user's chosen action (start new funding vs end current funding)
 */
export const ChangeFundingForm: React.FC<ChangeFundingFormProps> = ({
  fundingSpaces,
  reportingPeriods,
  enrollment,
  onSuccess,
}) => {
  const { accessToken } = useContext(AuthenticationContext);
  const [error, setError] = useState<string>();
  const [loading, setLoading] = useState(false);

  const onSubmit = (changeFunding: ChangeFunding) => {
    setLoading(true);
    apiPost(`enrollments/${enrollment.id}/change-funding`, changeFunding, {
      accessToken,
      jsonParse: false,
    })
      .then(() => {
        setError(undefined);
        setVisibleForm(undefined);
        onSuccess();
      })
      .catch((err) => {
        console.log(err);
        setError(
          'Unable to change funding. Make sure all required information is provided'
        );
      })
      .finally(() => {
        setLoading(false);
      });
  };

  const [visibleForm, setVisibleForm] = useState<'end' | 'start'>();

  const activeFunding = (enrollment.fundings || []).find(
    (f) => !f.lastReportingPeriod
  );
  return (
    <>
      {!!visibleForm && (
        <Card>
          <>
            <h3>
              {visibleForm === 'end' ? 'End current funding' : 'Change funding'}
            </h3>
            {error && <Alert type="error" text={error} />}
            <Form<ChangeFunding>
              id="change-funding-form"
              className="usa-form"
              data={{}}
              onSubmit={onSubmit}
            >
              {visibleForm === 'start' && (
                <FundingField<ChangeFunding>
                  fundingAccessor={(data) => data.at('newFunding')}
                  getEnrollment={() => enrollment}
                  reportingPeriods={reportingPeriods}
                  fundingSpaces={fundingSpaces}
                />
              )}
              {!!activeFunding && (
                <ReportingPeriodField<ChangeFunding>
                  accessor={(data) =>
                    data.at('oldFunding').at('lastReportingPeriod')
                  }
                  reportingPeriods={reportingPeriods.filter(
                    (rp) => rp.type === activeFunding.fundingSpace.source
                  )}
                  isLast={true}
                  label={`Last reporting period for current ${activeFunding.fundingSpace?.source} - ${activeFunding.fundingSpace?.time} funding`}
                  optional={visibleForm === 'start'}
                />
              )}
              <Button
                text="Cancel"
                appearance="outline"
                onClick={() => setVisibleForm(undefined)}
              />
              <FormSubmitButton
                text={
                  loading
                    ? visibleForm === 'end'
                      ? 'Ending funding...'
                      : 'Changing funding...'
                    : visibleForm === 'end'
                    ? 'End current funding'
                    : 'Change current funding'
                }
                disabled={loading}
              />
            </Form>
          </>
        </Card>
      )}
      <Card>
        <div className="display-flex flex-column flex-align-end">
          <div>
            {activeFunding && (
              <Button
                className="margin-right-4"
                text="End current funding"
                appearance="unstyled"
                onClick={() => setVisibleForm('end')}
              />
            )}
            <Button
              text="Start new funding"
              appearance="unstyled"
              onClick={() => setVisibleForm('start')}
            />
          </div>
        </div>
      </Card>
    </>
  );
};
