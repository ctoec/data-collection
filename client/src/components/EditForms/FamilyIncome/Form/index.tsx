import React, { useState } from 'react';
import { Button, Card } from '@ctoec/component-library';
import { propertyDateSorter } from '../../../../utils/dateSorter';
import { EditDeterminationForm } from './EditDeterminationForm';
import { RedeterminationForm } from './RedeterminationForm';
import { EditFormProps } from '../../types';

export const FamilyIncomeForm: React.FC<EditFormProps> = (props) => {
  const { child, onSuccess } = props;
  const [showRedeterminationForm, setShowRedeterminationForm] = useState(false);
  const [currentIsNew, setCurrentIsNew] = useState(false);

  if (!child) return <></>;

  const determinations = child.family.incomeDeterminations || [];
  const familyId = child.family.id;

  // Tracking variables for easy reference in element creation
  const sortedDeterminations = determinations.sort((a, b) =>
    propertyDateSorter(a, b, (det) => det.determinationDate, true)
  );
  const currentDetermination = sortedDeterminations[0];
  const pastDeterminations = sortedDeterminations.slice(1);

  return (
    <>
      <h2>
        Family income determination
          </h2>
      {showRedeterminationForm && (
        <>
          <h3>
            Redetermine family income
          </h3>
          <Card>
            <RedeterminationForm
              {...props}
              familyId={familyId}
              setIsNew={() => setCurrentIsNew(true)}
              hideForm={() => setShowRedeterminationForm(false)}
              onSuccess={onSuccess}
            />
          </Card>
        </>
      )}
      <div className="margin-top-1">
        <div className="display-flex align-center">
          <h3 className="font-sans-md margin-top-2 margin-bottom-2">
            {currentDetermination
              ? 'Current income determination'
              : 'No income information on record'}
          </h3>
          &nbsp;&nbsp;&nbsp;
          {!showRedeterminationForm && (
            <Button
              text="Redetermine income"
              appearance="unstyled"
              onClick={() => {
                setShowRedeterminationForm(true);
              }}
            />
          )}
        </div>
        <div>
          {currentDetermination && (
            <EditDeterminationForm
              {...props}
              determination={currentDetermination}
              familyId={familyId}
              isCurrent={true}
              isNew={currentIsNew}
              onSuccess={onSuccess}
            />
          )}
        </div>

        {pastDeterminations.length > 0 && (
          <>
            <div className="display-flex align-center">
              <h3>
                Past income determinations
              </h3>
            </div>
            <div>
              {pastDeterminations.map((determination) => (
                <EditDeterminationForm
                  {...props}
                  key={determination.id}
                  determination={determination}
                  familyId={familyId}
                  isCurrent={false}
                  isNew={false}
                  onSuccess={onSuccess}
                />
              ))}
            </div>
          </>
        )}
      </div>
    </>
  );
};
