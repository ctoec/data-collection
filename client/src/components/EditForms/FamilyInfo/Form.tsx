import React, { useState, useContext } from 'react';
import { Form, FormSubmitButton } from '@ctoec/component-library';
import { Family } from '../../../shared/models';
import { AddressFieldset, HomelessnessField } from './Fields';
import AuthenticationContext from '../../../contexts/AuthenticationContext/AuthenticationContext';
import { apiPut } from '../../../utils/api';
import { EditFormProps } from '../types';
import useIsMounted from '../../../hooks/useIsMounted';

/*
 * Functional component that allows a user to modify the address
 * information of a family nested within a Child object. The
 * family data is manipulated directly to avoid interactions
 * with the Child object that the family form doesn't need
 * to know about.
 */
export const FamilyInfoForm: React.FC<EditFormProps> = ({
  child,
  onSuccess,
  hideHeader = false,
}) => {
  const { accessToken } = useContext(AuthenticationContext);
  const isMounted = useIsMounted();
  const [saving, setSaving] = useState(false);

  if (!child) return <></>;

  const { family } = child;

  const onSubmit = (newState: Family) => {
    setSaving(true);
    apiPut(`families/${family.id}`, newState, { accessToken })
      .then(onSuccess)
      .catch((err) => {
        console.log(err);
      })
      .finally(() => (isMounted() ? setSaving(false) : null));
  };

  return (
    <div className="grid-container margin-top-2">
      {!hideHeader && (
        <h2 className="grid-row">Family Address</h2>
      )}
      <Form<Family>
        className="FamilyInfoForm"
        data={family}
        onSubmit={onSubmit}
        noValidate
        autoComplete="off"
      >
        <AddressFieldset />
        <h3 className="grid-row margin-top-4">Experiencing homelessness</h3>
        <HomelessnessField />
        <div className="grid-row margin-top-2">
          <FormSubmitButton
            text={saving ? 'Saving...' : 'Save edits'}
            disabled={saving}
          />
        </div>
      </Form>
    </div>
  );
};
