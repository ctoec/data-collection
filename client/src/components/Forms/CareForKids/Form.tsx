import React, { useState, useContext } from 'react';
import { Form, FormSubmitButton } from '@ctoec/component-library';
import { CareForKidsField } from './CareForKidsField';
import { EditFormProps } from '../types';
import AuthenticationContext from '../../../contexts/AuthenticationContext/AuthenticationContext';
import { Child } from '../../../shared/models';
import { apiPut } from '../../../utils/api';
/*
 * Basic functional component designed to allow user to edit
 * the Care For Kids field of a Child object.
 */
export const CareForKidsForm: React.FC<EditFormProps> = ({
  child,
  onSuccess,
}) => {
  const { accessToken } = useContext(AuthenticationContext);
  const [isSaving, setIsSaving] = useState(false);

  if (!child) return <></>;

  const onSubmit = (updatedChild: Child) => {
    setIsSaving(true);
    apiPut(`children/${child.id}`, updatedChild, { accessToken })
      .then(() => onSuccess())
      .catch((err) => {
        console.log(err);
      })
      .finally(() => setIsSaving(false));
  };

  return (
    <div className="grid-container margin-top-2">
      <h2 className="grid-row">Care 4 Kids</h2>
      <div>
        <Form<Child>
          className="CareForKidsForm"
          data={child}
          onSubmit={onSubmit}
          noValidate
          autoComplete="off"
        >
          <CareForKidsField />
          <div className="grid-row margin-top-2">
            <FormSubmitButton text={isSaving ? 'Saving...' : 'Save'} />
          </div>
        </Form>
      </div>
    </div>
  );
};
