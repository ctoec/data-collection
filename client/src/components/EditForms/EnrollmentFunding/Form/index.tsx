import React from 'react';
import { ChangeEnrollmentForm } from './ChangeEnrollmentForm';
import { EditEnrollmentForm } from './EditEnrollmentForm';
import { EditFundingForm } from './EditFundingForm';
import { ChangeFundingForm } from './ChangeFundingForm';
import { EditFormProps } from '../../types';
import { useSites } from '../../../../hooks/useSites';
import { useFundingSpaces } from '../../../../hooks/useFundingSpaces';
import { useReportingPeriods } from '../../../../hooks/useReportingPeriods';

export const EnrollmentFundingForm: React.FC<EditFormProps> = ({
  child,
  onSuccess,
  reportingPeriods: inputReportingPeriods,
}) => {
  // Get site options for new enrollments
  const { sites } = useSites();

  // Get fundingSpaces for new fundings
  const { fundingSpaces } = useFundingSpaces();

  // Get reporting periods (needed to update enrollments with fundings)
  const { reportingPeriods } = useReportingPeriods(inputReportingPeriods);

  // Separate enrollments into current (no end date) and past
  // (with end date). Either may not exist
  if (!child) {
    return <></>;
  }

  const enrollments = child.enrollments || [];
  const currentEnrollment = enrollments.find((e) => !e.exit);
  const pastEnrollments = currentEnrollment
    ? enrollments.filter((e) => e.id !== currentEnrollment.id)
    : enrollments;

  return (
    <>
      <ChangeEnrollmentForm
        reportingPeriods={reportingPeriods}
        fundingSpaces={fundingSpaces}
        childName={child.firstName}
        currentEnrollment={currentEnrollment}
        childId={child.id}
        sites={sites}
        onSuccess={onSuccess}
      />
      {currentEnrollment && (
        <>
          <h2 className="margin-top-4 margin-bottom-2 font-heading-md">
            Current enrollment
          </h2>
          <EditEnrollmentForm
            key="edit-current-enrollment"
            isCurrent={true}
            enrollment={currentEnrollment}
            onSuccess={onSuccess}
          />
          {currentEnrollment.fundings?.map((funding) => (
            <EditFundingForm
              key={funding.id}
              isCurrent={true}
              reportingPeriods={reportingPeriods}
              fundingSpaces={fundingSpaces}
              funding={funding}
              enrollment={currentEnrollment}
              onSuccess={onSuccess}
            />
          ))}
          <ChangeFundingForm
            fundingSpaces={fundingSpaces}
            reportingPeriods={reportingPeriods}
            enrollment={currentEnrollment}
            onSuccess={onSuccess}
          />
        </>
      )}
      {!!pastEnrollments.length && (
        <>
          <h2 className="margin-top-4 margin-bottom-2 font-heading-md">
            Past enrollments
          </h2>
          {pastEnrollments.map((enrollment) => (
            <>
              <EditEnrollmentForm
                key={enrollment.id}
                enrollment={enrollment}
                onSuccess={onSuccess}
              />
              {enrollment.fundings?.map((funding) => (
                <EditFundingForm
                  key={funding.id}
                  reportingPeriods={reportingPeriods}
                  fundingSpaces={fundingSpaces}
                  funding={funding}
                  enrollment={enrollment}
                  onSuccess={onSuccess}
                />
              ))}
            </>
          ))}
        </>
      )}
    </>
  );
};
