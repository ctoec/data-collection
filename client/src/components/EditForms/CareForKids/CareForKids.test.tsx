import React from 'react';
import {
  accessibilityTestHelper,
  snapshotTestHelper,
} from '../../../testHelpers';
import { CareForKidsForm } from './Form';
import { Family, Child } from '../../../shared/models';

const child = {
  id: '00000000-0000-0000-0000-000000000000',
  firstName: 'First',
  lastName: 'Last',
  family: {} as Family,
} as Child;

describe('EditRecord', () => {
  describe('CareForKids', () => {
    snapshotTestHelper(
      <CareForKidsForm
        child={child}
        onSuccess={jest.fn()}
        setAlerts={jest.fn()}
      />
    );
    accessibilityTestHelper(
      <CareForKidsForm
        child={child}
        onSuccess={jest.fn()}
        setAlerts={jest.fn()}
      />
    );
  });
});
