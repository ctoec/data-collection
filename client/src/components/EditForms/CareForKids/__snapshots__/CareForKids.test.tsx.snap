// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`EditRecord CareForKids matches snapshot 1`] = `
<DocumentFragment>
  <div
    class="grid-container margin-top-2"
  >
    <h2
      class="grid-row"
    >
      Receiving Care For Kids?
    </h2>
    <div>
      <form
        autocomplete="off"
        class="CareForKidsForm"
        novalidate=""
      >
        <fieldset
          class="grid-gap grid-row usa-fieldset"
          id="c4k-radio-group-fieldset"
        >
          <legend
            id="fieldset-legend-c4k-radio-group-fieldset"
          >
            <span
              class="usa-sr-only"
            >
              receives care for kids
            </span>
          </legend>
          <div
            class="grid-gap margin-top-3 grid-row flex-align-start"
          >
            <span>
              <div>
                <div
                  class="usa-radio"
                >
                  <input
                    class="usa-radio__input"
                    id="Yes"
                    name="careforkids"
                    type="radio"
                    value="Yes"
                  />
                  <label
                    class="usa-radio__label"
                    for="Yes"
                  >
                    Yes
                  </label>
                </div>
              </div>
            </span>
            <span>
              <div>
                <div
                  class="usa-radio"
                >
                  <input
                    checked=""
                    class="usa-radio__input"
                    id="No"
                    name="careforkids"
                    type="radio"
                    value="No"
                  />
                  <label
                    class="usa-radio__label"
                    for="No"
                  >
                    No
                  </label>
                </div>
              </div>
            </span>
          </div>
        </fieldset>
        <div
          class="grid-row margin-top-2"
        >
          <input
            class="usa-button"
            type="submit"
            value="Save"
          />
        </div>
      </form>
    </div>
  </div>
</DocumentFragment>
`;
